xds/src/main/java/com/linecorp/armeria/xds/BootstrapClusters.java

```diff
@@ -31,16 +29,21 @@
 final class BootstrapClusters implements SnapshotWatcher<ClusterSnapshot> {
 
     private final Map<String, ClusterSnapshot> clusterSnapshots = new HashMap<>();
+    private final Map<String, Cluster> edsClusters = new HashMap<>();
 
     BootstrapClusters(Bootstrap bootstrap, XdsBootstrapImpl xdsBootstrap) {
         if (bootstrap.hasStaticResources()) {
             final StaticResources staticResources = bootstrap.getStaticResources();
             for (Cluster cluster : staticResources.getClustersList()) {
-                checkArgument(cluster.hasLoadAssignment(),
-                              "Only fully static configurations are allowed for bootstrap clusters. " +
-                              "Violating cluster is %s", cluster);
-                // no need to clean this cluster up since it is fully static
-                StaticResourceUtils.staticCluster(xdsBootstrap, cluster.getName(), this, cluster);
+                if (cluster.hasLoadAssignment()) {
+                    // no need to clean this cluster up since it is fully static
+                    StaticResourceUtils.staticCluster(xdsBootstrap, cluster.getName(), this, cluster);
+                } else if (cluster.hasEdsClusterConfig()) {
+                    edsClusters.put(cluster.getName(), cluster);
```

Comment:
I forgot this point in the previous PR, but I believe `BootstrapClusters` and `BootstrapListeners` also needs to be cleaned up when `XdsBootstrapImpl` is closed (after all of the user-defined roots are closed)

Reply:
With the current implementation, `BootstrapClusters` doesn't need to be closed because the cluster is closed when its primer is closed.
Do you want me to fetch the load assignment in the constructor of the `BootstrapClusters` before the cluster is actually used?

Reply:
Also, `BootstrapListeners` does not have any resources for cleanup.
The `ListenerResourceNode` is closed when the `ListenerRoot` is closed. Please correct me if I'm wrong. 😉 
https://github.com/line/armeria/blob/7e6cfbacd8af53031f02f4b7cb49d3e7c42dc97f/xds/src/main/java/com/linecorp/armeria/xds/ListenerRoot.java#L51

Reply:
Oops, I guess I skimmed the PR too quickly.  I think you're right!